File: test/fixtures/package_project/src/kernels.ml
open Loom

let[@loom.entry] saxpy (a : float) (x : Tensor1.t) (y : Tensor1.t) =
  Tensor1.map2 (fun xi yi -> a *. xi +. yi) x y

let[@loom.entry] relu (x : Tensor1.t) =
  Tensor1.map (fun xi -> if xi > 0. then xi else 0.) x

let[@loom.entry] dot (x : Tensor1.t) (y : Tensor1.t) =
  let products = Tensor1.map2 (fun xi yi -> xi *. yi) x y in
  Tensor1.reduce_sum products

File: test/fixtures/python_package_project/kernels.py
import loom


@loom.entry
def saxpy(a: float, x: loom.Tensor1, y: loom.Tensor1) -> loom.Tensor1:
    return loom.Tensor1.map2(lambda xi, yi: a * xi + yi, x, y)


@loom.entry
def relu(x: loom.Tensor1) -> loom.Tensor1:
    return loom.Tensor1.map(lambda xi: xi if xi > 0.0 else 0.0, x)


@loom.entry
def dot(x: loom.Tensor1, y: loom.Tensor1) -> float:
    products = loom.Tensor1.map2(lambda xi, yi: xi * yi, x, y)
    return loom.Tensor1.reduce_sum(products)
